### PR TITLE
[optimizer] fix wd in sgd

### PR DESCRIPTION
**Pull request by antinucleon**

### CODE REVIEW

**Comment by piiswrong** [comment]
To be clear, the only problem with ccsgd is pickle right? I'll try to fix it.

**Comment by antinucleon** [comment]
@piiswrong First is pickle, it is easy, only need to add __getstate__ to deal with pointer. The second is we only need to do weight decay on weight, this is a little headache for ccSGD

**Comment by piiswrong** [comment]
@antinucleon SGD is also decaying bias right? should we fix this?

**Comment by antinucleon** [comment]
I have fixed SGD. According to Naiyan we should fix it

**Comment by piiswrong** [comment]
Relying on name still feels unreliable. Plus with this patch by default no weights will be decayed with custom model.

**Comment by antinucleon** [comment]
I made a mistake after Naiyan's review. By default if we don't pass arg_name in, there will be wd for all param. For custom model, I find a way to make pickle possible for distributed training, so maybe we can write custom sgd for custom model. I am going to fix the wd default bug now.
